### PR TITLE
Handwritten resource and data source tutorial

### PR DESCRIPTION
**Pull request by ScottSuarez**
closes https://github.com/hashicorp/terraform-provider-google/issues/11713
closes https://github.com/hashicorp/terraform-provider-google/issues/11962

I've implemented the resource guide in full and a partial guide for the data sources.

Let me know if you want me to expand on anything here. 
**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/README.md [hunk 1]
```diff
@@ -27,9 +28,134 @@ Handwritten resources like [google_container_cluster](https://registry.terraform
...
+are no longer needed you can remove the `.erb` extension.
+
+#### Create, Read, Update, Delete
+As far as terraform schema is concerned these are the functions we
+need to provide for terraform to be able to provision and delete
+resources. In editing any fields you'll likely be adding functionality to
+these functions or implementing them wholesale.
+
+
+#### Expanders and Flatteners
```
Comment (shuyama1) [inline mmv1/third_party/terraform/README.md:59]: Maybe add an example here?

**Comment by ScottSuarez** [comment]
Hey Shuya, 

I implemented most of your suggestions. Sorry for the delay here. Let me know if anything else needs to be done!

**Review by shuyama1 — APPROVED** [review]
LGTM! Thank you so much for updating this!

**Comment by c2thorn** [comment]
Should this be merged? I have someone asking for datasource documentation
@shuyama1 @ScottSuarez

**Pull request merged**